### PR TITLE
Enable support for PEM keystores and truststores

### PR DESCRIPTION
**Pull request by jrcast**
Enable support for PEM keystores and truststores

**NOTE: Requires `PEM Key Manager` and `PEM Trust Manager` enabled in the docker image.**

### CODE REVIEW

File: controllers/sts.go [hunk 1]
```diff
@@ -216,15 +228,22 @@ func createDSStatefulSet(ds *directoryv1alpha1.DirectoryService, sts *apps.State
...
+									SubPath:   "master-key-pair-combined.pem",
 								},
 								{
 									Name:      "cloud-backup-credentials",
 									MountPath: "/var/run/secrets/cloud-credentials-cache/",
 								},
 								{
 									Name:      "pem-trust-certs",
-									MountPath: "/opt/opendj/pem-trust-certs",
+									MountPath: "/opt/opendj/pem-trust-directory/trust.pem",
```
Comment (maxres-fr) [inline controllers/sts.go:245]: s/trust/truststore seems slightly better to me.

**Review by maxres-fr — APPROVED** [review]

File: controllers/sts.go [hunk 2]
```diff
@@ -216,15 +228,22 @@ func createDSStatefulSet(ds *directoryv1alpha1.DirectoryService, sts *apps.State
 								},
 								{
 									Name:      "secrets",
-									MountPath: "/opt/opendj/secrets",
+									MountPath: "/opt/opendj/pem-keys-directory/ssl-key-pair",
+									SubPath:   "ssl-key-pair-combined.pem",
+								},
+								{
+									Name:      "secrets",
+									MountPath: "/opt/opendj/pem-keys-directory/master-key",
```
[the remark this answers is not in the document]
Comment (jrcast) [reply controllers/sts.go:236]: Initially, I had `master-key-pair`. However, DS was looking for `master-key` alias so I had to rename it to that. 
I didn't want to add code in setup.sh just to change this with the fear it would just introduce bugs. I just decided to use the `master-key` default name instead

[the remark this answers is not in the document]
Comment (lee-baines) [reply controllers/sts.go:236]: ok no problem. Just checking that you didn't just miss it.

File: controllers/sts.go [hunk 3]
```diff
@@ -216,15 +228,22 @@ func createDSStatefulSet(ds *directoryv1alpha1.DirectoryService, sts *apps.State
...
+									SubPath:   "master-key-pair-combined.pem",
 								},
 								{
 									Name:      "cloud-backup-credentials",
```
Comment (wstrange) [inline controllers/sts.go:240]: strongly considering deprecating this - but we can do that in another PR

**Review by wstrange — APPROVED** [review]
LGTM. I assume PR  build is failing because the docker image is not compatible?

We need to think about how that can be fixed
